plugins/couchbase/couchbaseembed.hpp

```diff
@@ -0,0 +1,390 @@
+                break;
+            case Deserialization_NOT_A_NUMBER:
+                failx("Deserialization error (%s): non-numeric characters found in numeric conversion: '%s'", targetype, culpritvalue);
+                break;
+            case Deserialization_OVERFLOW:
+                failx("Deserialization error (%s): number too large to be represented by receiving value", targetype);
+                break;
+            case Deserialization_UNDERFLOW:
+                failx("Deserialization error (%s): number too small to be represented by receiving value", targetype);
+                break;
+            default:
+                typeError(targetype, culpritvalue);
+                break;
+        }
+    }
+
+    class CouchbaseRowStream : public RtlCInterface, implements IRowStream
+    {
+    public:
+        CouchbaseRowStream(IEngineRowAllocator* _resultAllocator, Couchbase::Query * cbaseQuery);
+        virtual ~CouchbaseRowStream();
+
+        RTLIMPLEMENT_IINTERFACE
+        virtual const void* nextRow();
+        virtual void stop();
+    private:
+        Couchbase::Query *              m_CouchBaseQuery;   //!< pointer to couchbase query (holds results and metadata)
+
+        Linked<IEngineRowAllocator>     m_resultAllocator;  //!< Pointer to allocator used when building result rows
+        bool                            m_shouldRead;       //!< If true, we should continue trying to read more messages
+        StringArray                     m_Rows;             //!< Local copy of result rows
+        __int64                         m_currentRow;       //!< Current result row
+
+    };
+
+    class CouchbaseConnection : public CInterface
+    {
+    public:
+        inline CouchbaseConnection(bool useSSL, const char * host, unsigned port, const char * bucketname, const char * user, const char * password, const char * connOptions)
+        {
+            m_connectionString.setf("couchbase%s://%s/%s%s", useSSL ? "s" : "", host, bucketname, connOptions);
+            m_pCouchbaseClient = new Couchbase::Client(m_connectionString.str());//USER/PASS still needed
+            m_pQuery = nullptr;
+        }
+
+        inline ~CouchbaseConnection()
+        {
+        }
+
+        inline void connect()
+        {
+            m_connectionStatus = m_pCouchbaseClient->connect();
+            if (!m_connectionStatus.success())
+                failx("Failed to connect to couchbase instance: %s Reason: %s", m_connectionString.str(), m_connectionStatus.description());
+        }
+
+        Couchbase::Query * query(Couchbase::QueryCommand * qcommand);
+
+    private:
+        StringBuffer m_connectionString;
+        Couchbase::Client * m_pCouchbaseClient;
+        Couchbase::Status  m_connectionStatus;
+        Couchbase::Query  * m_pQuery;
+
+        CouchbaseConnection(const CouchbaseConnection &);
+    };
+
+    class CouchbaseRowBuilder : public CInterfaceOf<IFieldSource>
```

Comment:
The code for this class feels like it should be included in the .cpp file instead of the header.  Was there a dependency issue?